neps/nep-0393.md

```diff
@@ -0,0 +1,552 @@
+
+This becomes especially important for proof-of-human stamps that can only be issued once per user. Few safeguards against misuse of SBT transfer (soul transfer or recovery) are contemplated:
+
+1. Users cannot recover an SBT by themselves. The issuer, a DAO or a smart contract (e.g. multisig) dedicated to manage the recovery should be assigned.
+2. Whenever a soul transfer is triggered then the SBT registry emits `SoulTransfer` and `Ban` events. The latter signals that the account can't host nor receive any SBT in the future. It creates an inherit cost for such action: the account identity is burned.
+3. `SoulTransfer` event can also trigger similar actions in other registries (specification for this is out of the scope of this NEP).
+4. The recovery function is additional economical risk preventing account trading: user should always be able to recover his SBT, and move to another, not banned account.
+
+SBT recover MUST not trigger `SoulTransfer` nor `Ban`: malicious issuer could compromise the system by faking the token recovery and take over all other SBTs from a user. Only the owner of the account can make a Soul Transfer transaction and merge 2 accounts they owns.
+
+Soulbound tokens can have an _expire date_. This is useful for tokens which are related to real world certificates with expire time, or social mechanisms (e.g. community membership). Such tokens SHOULD have an option to be renewable. Examples include mandatory renewal with a frequency to check that the owner is still alive, or renew membership to a DAO that uses SBTs as membership gating.
+
+An issuer can provide an _sbt revocation_ in his contract (for example: when a related certificate or membership should be revoked, when an issuer finds out that there was an abuse or a scam, etc...). Registry, when receiving `sbt_revoke` request from an issuer must always emit the `Revoke` event. Registry must only accept revoke requests from a valid issuer, and only revoke tokens from that issuer. If `burn=true` is set in the request, then the token should be burned and additionally `Burn` event must be emitted. Otherwise (when `burn=false`) the registry must update token metadata and set expire date in the past. Registry must not ban nor emit `Ban` event when revoking a contract. That would create an attack vector, when a malicious registry would thread the registry by banning accounts.
+
+A registry can emit a `Ban` event without doing soul transfer. Handling it depends on the registry governance or registry use cases. One example is to use social governance to identify fake accounts (like bots) - in that case the registry should allow to emit `Ban` and block a scam soul and block future transfers.
+NOTE: SBT smart contract (issuer) can have it's own list of blocked accounts or allowed only accounts.
+
```

Comment:
I feel like it should be structured like this (definitions to definitions, constraints to constraints).

```suggestion
1. **soul transfer** when a user needs to merge his accounts (e.g. they started with few different account but later decides to merge them to increase an account reputation). Soul transfer is different than a token transfer. The standard forbids a traditional token transfer functionality, where a user can transfer individual tokens. That being said, a registry can have extension functions for more advanced scenarios, which could require a governance mechanism.
   * Whenever a soul transfer is triggered then the SBT registry emits `SoulTransfer` and `Ban` events. The latter signals that the account can't host nor receive any SBT in the future. It creates an inherit cost for such action: the account identity is burned.
   * `SoulTransfer` event can also trigger similar actions in other registries (specification for this is out of the scope of this NEP).
2. **recoverability** in case a user's private key is compromised due to extortion, loss, etc.
   * The recovery function is additional economical risk preventing account trading: user should always be able to recover his SBT, and move to another, not banned account.
3. A registry can emit a `Ban` event _without_ doing soul transfer.
   * Handling it depends on the registry governance or registry use cases. One example is to use social governance to identify fake accounts (like bots) - in that case the registry should allow to emit `Ban` and block a scam soul and block future transfers.
    NOTE: SBT smart contract (issuer) can have it's own list of blocked accounts or allowed only accounts.

This becomes especially important for proof-of-human stamps that can only be issued once per user.

Few safeguards against misuse of SBT transfer (soul transfer or recovery) are contemplated:

1. Users cannot recover an SBT by themselves. The issuer, a DAO or a smart contract (e.g. multisig) dedicated to manage the recovery should be assigned.
2. SBT recover MUST not trigger `SoulTransfer` nor `Ban`: malicious issuer could compromise the system by faking the token recovery and take over all other SBTs from a user. Only the owner of the account can make a Soul Transfer transaction and merge 2 accounts they owns.
3. Soulbound tokens can have an _expire date_. This is useful for tokens which are related to real world certificates with expire time, or social mechanisms (e.g. community membership). Such tokens SHOULD have an option to be renewable. Examples include mandatory renewal with a frequency to check that the owner is still alive, or renew membership to a DAO that uses SBTs as membership gating.
4. An issuer can provide an _sbt revocation_ in his contract (for example: when a related certificate or membership should be revoked, when an issuer finds out that there was an abuse or a scam, etc...). Registry, when receiving `sbt_revoke` request from an issuer must always emit the `Revoke` event. Registry must only accept revoke requests from a valid issuer, and only revoke tokens from that issuer. If `burn=true` is set in the request, then the token should be burned and additionally `Burn` event must be emitted. Otherwise (when `burn=false`) the registry must update token metadata and set expire date in the past. Registry must not ban nor emit `Ban` event when revoking a contract. That would create an attack vector, when a malicious registry would thread the registry by banning accounts.

```

Reply:
@robert-zaremba Let me try to explain my point, my last try before you reject :) My perception of your narration is that you mixing up _safeguards/constraints_ and _functional requirements_ across the paragraphs:

1. [Constraint/safeguard] Users cannot recover an SBT by themselves...
2. [Functional requirement about transferability] Whenever a soul transfer is triggered then the SBT registry emits `SoulTransfer` and `Ban` events...
3. [Out-of-scope note] `SoulTransfer` event can also trigger similar actions... (specification for this is out of the scope of this NEP).
4. [Functional requirement/constraint about recoverability] .... user should always be able to recover his SBT, and move to another, not banned account.

[Constraint of recoverability functional requirement] SBT recover MUST not trigger `SoulTransfer` nor `Ban`...
[Functional requirement] Soulbound tokens can have an _expire date_. This is useful for tokens which...
[Functional requirement] An issuer can provide an _sbt revocation_ in his contract...
[Functional requirement  A registry can emit a `Ban` event without doing soul transfer. Handling it depends on the registry governance or registry use cases....
[Note :) ] NOTE: SBT smart contract....

In my example I grouped "kinda functional requirements" under 2 groups: `transferability` and `recoverability`, and safeguards go in a separate list.

Reply:
I've read both your proposals a few times over. I can see benefits of both structures. It boils down to style, as the content is the same in both proposals. Robert's argument makes sense, there are some features and safeguards that span both soultransfer and recovery. My vote to keep it as is for this reason and because a lot of people have already read, commented and familiarized themselves based on the original storyline.